src/lib.rs

```diff
@@ -340,7 +340,17 @@ mod hyper_impl {
             self: Pin<&mut Self>,
             cx: &mut Context<'_>,
         ) -> Poll<Option<Result<Self::Conn, Self::Error>>> {
-            self.poll_next(cx)
+            match self.poll_next(cx) {
+                Poll::Ready(Some(res)) => match res {
+                    Ok(val) => Poll::Ready(Some(Ok(val))),
+                    Err(e) => {
+                        println!("error {}", e);
+                        Poll::Pending
```

Comment:
This should be `Poll::Ready(Some(Err(...)))` not pending

Reply:
I still encountered issues with the hyper server stopping when I made that change, but returning Poll::Pending did not have that problem.